libs/lakebase/connect.go

```diff
@@ -0,0 +1,104 @@
+package lakebase
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/databricks/cli/libs/cmdctx"
+	"github.com/databricks/cli/libs/exec"
+	"github.com/databricks/databricks-sdk-go/service/database"
+	"github.com/google/uuid"
+)
+
+func Connect(ctx context.Context, databaseInstanceName string, extraArgs ...string) error {
+	fmt.Printf("Connecting to Databricks Database Instance %s ...\n", databaseInstanceName)
+
+	w := cmdctx.WorkspaceClient(ctx)
+
+	// get user:
+	user, err := w.CurrentUser.Me(ctx)
+	if err != nil {
+		fmt.Println("Error getting current user: ", err)
+		return err
+	}
+
+	// get database:
+	db, err := w.Database.GetDatabaseInstance(ctx, database.GetDatabaseInstanceRequest{
+		Name: databaseInstanceName,
+	})
+	if err != nil {
+		fmt.Println("Error getting Database Instance: ", err)
+		fmt.Println("Does the database instance exist?")
+		return err
+	}
+
+	fmt.Println("Database status: ", db.State)
+	fmt.Println("Database postgres version: ", db.PgVersion)
+
+	// get credentials:
+	cred, err := w.Database.GenerateDatabaseCredential(ctx, database.GenerateDatabaseCredentialRequest{
+		InstanceNames: []string{databaseInstanceName},
+		RequestId:     uuid.NewString(),
+	})
+	if err != nil {
+		fmt.Println("Error getting database credentials: ", err)
+		return err
+	}
+	fmt.Println("Successfully fetched database credentials")
+
+	// Get current working directory
+	dir, err := os.Getwd()
+	if err != nil {
+		fmt.Printf("Error getting working directory: %v\n", err)
+		return err
+	}
+
+	// Check if database name and port are already specified in extra arguments
+	hasDbName := false
+	hasPort := false
+	for _, arg := range extraArgs {
+		if arg == "-d" || strings.HasPrefix(arg, "--dbname=") {
```

Comment:
It can be more robust to make these flags at the cobra level. Parsing whether the flag is set can be brittle. If we make these flags in our CLI then we can always append to `extraArgs`

For example I found this example as well online:
```
psql "sslmode=require host=<db-address> dbname=<db-name> user=<username>"
```
link: https://hasura.io/blog/top-psql-commands-and-flags-you-need-to-know-postgresql

If we take this approach, we should confirm what happens if a user sets --dbname in `extraArgs` as well.